src/components/Avatar/Avatar.tsx

```diff
@@ -0,0 +1,179 @@
+     * */
+    size?: number | string;
+};
+
+const Avatar = (
+    {
+        source,
+        label,
+        iconName,
+        iconType,
+        color,
+        style,
+        size: sizeProp,
+        iconProps,
+        imageProps,
+        labelProps,
+        ...rest
+    }: Props,
+    ref: any,
+) => {
+    const { View } = useMolecules();
+
+    const componentStyles = useComponentStyles('Avatar', [
+        style,
+        color ? { backgroundColor: color } : {},
+        sizeProp ? { size: sizeProp } : {},
+    ]);
+
+    const { containerStyle, iconStyle, imageStyle, labelStyle, size } = useMemo(() => {
+        const {
+            image: _imageStyle,
+            icon: _iconStyle,
+            label: _labelStyle,
+            size: _size,
+            ...restStyles
+        } = componentStyles;
+
+        return {
+            containerStyle: [restStyles, { width: _size, height: _size }],
+            imageStyle: [_imageStyle, imageProps?.style],
+            iconStyle: [_iconStyle, iconProps?.style],
+            labelStyle: [_labelStyle, { fontSize: _size * 0.5 }, labelProps?.style],
+            size: _size,
+        };
+    }, [componentStyles, iconProps?.style, imageProps?.style, labelProps?.style]);
+
+    return (
+        <View ref={ref} {...rest} style={containerStyle}>
+            <AvatarInner
+                source={source}
+                imageProps={imageProps}
+                iconProps={iconProps}
+                iconStyle={iconStyle}
+                size={size}
+                iconName={iconName}
+                label={label}
+                labelProps={labelProps}
+                labelStyle={labelStyle}
+                iconType={iconType}
+                imageStyle={imageStyle}
+            />
+        </View>
+    );
+};
+
+type AvatarInnerProps = Pick<
+    Props,
+    'iconName' | 'iconType' | 'iconProps' | 'imageProps' | 'labelProps' | 'label' | 'source'
+> & {
+    imageStyle?: ImageProps['style'];
+    iconStyle?: IconProps['style'];
+    labelStyle?: TextProps['style'];
+    // at this point size will always be a number
+    size: number;
+};
+
+const AvatarInner = memo(
+    ({
+        imageStyle,
+        source,
+        iconStyle,
+        labelStyle,
+        imageProps,
+        iconProps,
+        labelProps,
+        iconName,
+        iconType,
+        label: labelProp,
+        size,
+    }: AvatarInnerProps) => {
+        const { Image, Icon, Text } = useMolecules();
+        const [isImageFailed, setIsImageFailed] = useState(false);
+
+        const normalizeLabel = useMemo(() => {
+            const { firstWord, lastWord } = extractFirstAndLastWord(labelProp || '');
+
+            return `${firstWord[0] || ''}${lastWord[0] || firstWord[1] || ''}`;
+        }, [labelProp]);
+
+        const onImageLoadFailed = useCallback(
+            (e: NativeSyntheticEvent<ImageErrorEventData>) => {
+                imageProps?.onError?.(e);
+
+                setIsImageFailed(true);
```

Comment:
If the image URL changes, isImageFailed should be reset to false.

Reply:
imageProps could be an object, add only onError function to the dependency array